packages/loader/container-loader/src/container.ts

```diff
@@ -174,43 +197,22 @@ export async function waitContainerToCatchUp(container: IContainer) {
     }
 
     return new Promise<boolean>((resolve, reject) => {
-        const deltaManager = container.deltaManager;
-
         container.on("closed", reject);
 
-        const waitForOps = () => {
-            assert(container.connectionState !== ConnectionState.Disconnected,
-                0x0cd /* "Container disconnected while waiting for ops!" */);
-            const hasCheckpointSequenceNumber = deltaManager.hasCheckpointSequenceNumber;
-
-            const connectionOpSeqNumber = deltaManager.lastKnownSeqNumber;
-            assert(deltaManager.lastSequenceNumber <= connectionOpSeqNumber,
-                0x266 /* "lastKnownSeqNumber should never be below last processed sequence number" */);
-            if (deltaManager.lastSequenceNumber === connectionOpSeqNumber) {
-                resolve(hasCheckpointSequenceNumber);
-                return;
-            }
-            const callbackOps = (message: ISequencedDocumentMessage) => {
-                if (connectionOpSeqNumber <= message.sequenceNumber) {
-                    resolve(hasCheckpointSequenceNumber);
-                    deltaManager.off("op", callbackOps);
-                }
-            };
-            deltaManager.on("op", callbackOps);
-        };
+        //* I don't fully understand this comment
```

Comment:
I think it points out to a fact that if we do not have checkpointSequenceNumber, then we can at least leverage join op sequence number for same purposes (for "write" connections).

What this code does not do a good job is dealing with disconnects while waiting for last op. I think it's worth opening a bug to incorporate that.